certbot/main.py

```diff
@@ -859,6 +861,45 @@ def plugins_cmd(config, plugins):
     logger.debug("Prepared plugins: %s", available)
     notify(str(available))
 
+def enhance(config, plugins):
+    """Add security enhancements to existing configuration
+
+    :param config: Configuration object
+    :type config: interfaces.IConfig
+
+    :param plugins: List of plugins
+    :type plugins: `list` of `str`
+
+    :returns: `None`
+    :rtype: None
+
+    """
+    supported_enhancements = ["hsts", "redirect", "uir", "staple"]
+    # Check that at least one enhancement was requested on command line
+    if not any([getattr(config, enh) for enh in supported_enhancements]):
+        msg = ("Please specify one or more enhancement types to configure. To list "
+               "the available enhancement types, run:\n\n%s --help enhance\n")
+        logger.warning(msg, sys.argv[0])
+        raise errors.MisconfigurationError("No enhancements requested, exiting.")
+
+    try:
+        installer, _ = plug_sel.choose_configurator_plugins(config, plugins, "enhance")
+    except errors.PluginSelectionError as e:
+        return str(e)
+
+    if not config.certname:
+        config.certname = cert_manager.get_certnames(config, "enhance",
+                                                     allow_multiple=False)[0]
+    cert_domains = cert_manager.domains_for_certname(config, config.certname)
+    domain_question = ("Which domain names would you like to enable the selected "
+                       "enhancements for?")
+    domains = display_ops.choose_values(cert_domains, domain_question)
```

Comment:
`IDisplay` has become horribly complex over the years (see #3915), but this call causes `IDisplay.checklist` to be called with `force_interactive=True`. I don't think this is what you want here as rather than requiring interactivity, I believe [the plan](https://github.com/certbot/certbot/issues/5551#issuecomment-377004693) was to use all the domains in the certificate when run non-interactively. Currently if you run this with `--non-interactive`, you get:
```
MissingCommandlineFlag: Missing command line flag or config entry for this setting:
Which domain names would you like to enable the selected enhancements for?
```
I think you want to remove the `force_interactive` parameter (or set it to `False`) and include `default` set to `cert_domains` in the call to `IDisplay.checklist`. This may mean you don't add the generic `choose_values` function or you add these values as additional parameters to the function.

Reply:
It's now using `cert_manager.domains_for_certname` if noninteractive mode is set.